lib/utils.pm

```diff
@@ -753,17 +753,19 @@ sub handle_login {
         }
         type_string "root\n";
     }
-    if (get_var('DM_NEEDS_USERNAME') and !get_var('ROOTONLY')) {
-        type_string "$username\n";
-    }
-    if (check_var('DESKTOP', 'gnome') || (check_var('DESKTOP', 'lxde') && check_var('VERSION', '42.1'))) {
-        # DMs in condition above have to select user
-        if ((is_sle && sle_version_at_least('15')) || (is_leap && leap_version_at_least('15.0')) || is_tumbleweed) {
-            assert_and_click "displaymanager-$username";
-            record_soft_failure 'bgo#657996 - user account not selected by default, have to use mouse to login';
+    else {
+        if (get_var('DM_NEEDS_USERNAME')) {
+            type_string "$username\n";
         }
-        else {
-            send_key 'ret';
+        if (check_var('DESKTOP', 'gnome') || (check_var('DESKTOP', 'lxde') && check_var('VERSION', '42.1'))) {
```

Comment:
I don't understand how this branch should be triggered. See https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4120#pullrequestreview-88160520

Reply:
If I'm reading it correctly, it's only triggered for any OS when DESKTOP is gnome, or for any OS when DESKTOP is lxde and VERSION is 42.1.

In the case of the SLES4SAP 15 tests, it's entering into that block just because DESKTOP is gnome. I guess it's not that visible in SLES' tests because those tests create an unprivileged user during installation so they don't fail, but as you can see here: https://openqa.suse.de/tests/1374890#step/first_boot/5, a soft failure is being recorded (which now I am not sure if it's a false positive, after reading the bug description here: https://bugzilla.gnome.org/show_bug.cgi?id=657996)